tests/components/modbus/test_climate.py

```diff
@@ -625,6 +674,95 @@ async def test_service_climate_update(
     assert hass.states.get(ENTITY_ID).state == result
 
 
+@pytest.mark.parametrize(
+    ("do_config", "result", "register_words"),
+    [
+        (
+            {
+                CONF_CLIMATES: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_TARGET_TEMP: 116,
+                        CONF_ADDRESS: 117,
+                        CONF_SLAVE: 10,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_DATA_TYPE: DataType.INT32,
+                        CONF_HVAC_ACTION_REGISTER: {
+                            CONF_ADDRESS: 118,
+                            CONF_HVAC_ACTION_VALUES: {
+                                CONF_HVAC_ACTION_IDLE: 0,
+                                CONF_HVAC_ACTION_HEATING: 1,
+                            },
+                        },
+                    },
+                ]
+            },
+            HVACAction.HEATING,
+            [0x01],
+        ),
+        (
+            {
+                CONF_CLIMATES: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_TARGET_TEMP: 116,
+                        CONF_ADDRESS: 117,
+                        CONF_SLAVE: 10,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_DATA_TYPE: DataType.INT32,
+                        CONF_HVAC_ACTION_REGISTER: {
+                            CONF_ADDRESS: 118,
+                            CONF_HVAC_ACTION_VALUES: {
+                                CONF_HVAC_ACTION_COOLING: 0,
+                                CONF_HVAC_ACTION_HEATING: 1,
+                            },
+                        },
+                    },
+                ]
+            },
+            HVACAction.COOLING,
+            [0x00],
+        ),
+        (
+            {
+                CONF_CLIMATES: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_TARGET_TEMP: 116,
+                        CONF_ADDRESS: 117,
+                        CONF_SLAVE: 10,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_DATA_TYPE: DataType.INT32,
+                        CONF_HVAC_ACTION_REGISTER: {
+                            CONF_ADDRESS: 118,
+                            CONF_HVAC_ACTION_VALUES: {
+                                CONF_HVAC_ACTION_OFF: 0,
+                                CONF_HVAC_ACTION_DRYING: 1,
+                            },
+                        },
+                    },
+                ]
+            },
+            HVACAction.DRYING,
+            [0x01],
+        ),
+    ],
+)
+async def test_service_climate_action_update(
```

Comment:
can we move this test after `test_hvac_onoff_coil_update`? Because both `test_hvac_onoff_coil_update` and the test above (`test_service_climate_update`) test the same thing (main entity state).